Add selector to interrogate categories

vladmandic opened the pull request:
switch from predefined `artists.csv` in general is good, but it introduces several problems

user may be interested in specific categories, but not others - for example, i'd like to see **mediums**, but i'm not interested in **artists**  

this pr:
- introduces `opts.interrogate_clip_categories` which is a string list of categories to use
- enhances caching so loaded list is refreshed if user settings change
- modifies how categories are downloaded
- adds torch garbage collect inside second categories lookup

btw, loading all categories if `interrogate_clip_dict_limit == 0` will guaranteed result in `torch.cuda.OutOfMemoryError: CUDA out of memory` as its impossible to load all categories at once  

and clipping the dict may not be what user exepects since it will result in suboptimal results (e.g. suggested artist will come from a much smaller list of artists)  

we may want to set sane maximum for `clip_dict_limit` and disable no-limit value and document what does it mean

vladmandic committed:
improve interrogate

AUTOMATIC1111 commented:
The user is meant to edit those files himself; the download only happens in the directory does not exist. Since there already is a way to edit files for LIP interrogate, I don't want to have a setting for it or pre-define category types. The instruction for those files is already on wiki.

AUTOMATIC1111 commented:
If you make the setting to be able to select out of existing files in the directory, that would be nice.
Also consider if dropdown with multiselect works better than checkboxes.

vladmandic committed:
Merge branch 'AUTOMATIC1111:master' into interrogate

vladmandic committed:
add option to skip interrogate categories

vladmandic commented:
> If you make the setting to be able to select out of existing files in the directory, that would be nice. Also consider if dropdown with multiselect works better than checkboxes.

Updated PR:

- if folder does not exist sets default clip category list to "artists", "flavors", "mediums", "movements" and downloads content
- adds option `interrogate_clip_skip_categories` which works as **exclude** list so all newly added files are automatically included
- list can be refreshed on-the-fly
- if user selection is changed, content is reloaded (no restart required)
- modified file enum so only `*.txt` files are looked at

*note*: i had some issues making `OptionInfo` work with `gr.Dropdown` with `multiselect` property (`update` method complains about invalid property) so i've left it as `gr.CheckboxGroup` for now
